app/controllers/foods_controller.rb

```diff
@@ -3,11 +3,32 @@ def index
     @foods = Food.all
   end
 
-  def new; end
+  def new
+    @food = Food.new
+  end
 
   def create
-    @food = Food.new(food_params)
+    food = current_user.foods.new(food_params)
+
+    if food.save
+      redirect_to foods_path, notice: 'New Food was successfully added.'
+    else
+      flash[:alert] = 'New Food adding Failed. Please try again.'
+    end
+  end
+
+  def destroy
+    food = Food.find(params[:id])
+
+    if food.destroy
+      flash[:notice] = 'Food was successfully deleted.'
+    else
+      flash[:alert] = 'Food deleting Failed. Please try again.'
+    end
+    redirect_back(fallback_location: root_path)
```

Comment:
Please make sure that a user can only delete foods added by themselves.
You can use something like `if food.user == current_user`